internal/beater/java_attacher/java_attacher.go

```diff
@@ -167,6 +177,71 @@ func (j *JavaAttacher) Run(ctx context.Context) error {
 	}
 }
 
+func (j *JavaAttacher) deleteTempDir(tempDir string) {
+	func(name string) {
+		err := os.RemoveAll(name)
+		if err != nil {
+			j.logger.Errorf("failed to delete Java attacher temp dir at %v: %v", tempDir, err)
+		}
+	}(tempDir)
+}
+
+// Creates a temp dir for the java attacher jar with the attacher jar in it, both with 0755 access permissions.
+// The assumption is that the default temp dir will be readable for all users.
+// However, there may be systems in which the temp directory is user-specific,
+// in which case we'll need to modify this function to run a temp dir search per user,
+// maybe by externally executing `echo XXX` and reading the output, relying on environment
+// variables as implemented in os.TempDir, or looking in known paths.
+// In such case, we would switch to using a cache of temp-agent-dir per user.
+func (j *JavaAttacher) createAttacherTempDir() (string, error) {
```

Comment:
I am planning to change that very soon, so that there will be a directory per user with the proper permissions, including write. This way, in combination with `os.Chown`, we can cache the attacher jar instead of downloading over and over for any attachment.

Reply:
Let me see if I understand correctly:
 - the bundled attacher will download the full attacher from Maven, and we would like that to be cached
 - this needs to be cached per user (because it is downloaded by a process running as that user)

Does that mean the bundled attacher needs to be copied for each user though? Does the directory holding the bundled attacher and the directory in which the attacher is downloaded have to be the same?

Reply:
I will change as proposed and we can rethink if we want to introduce agent jar caching as described [above](https://github.com/elastic/apm-server/pull/8803#discussion_r942079198).

For agent jar caching, we need the attacher process to run as the target (application) user, and make sure the current directory is only writable to the current user. The way to ensure that is create a tmp dir per user, make this user the owner of that dir and set its mode to `0700`.

Reply:
Implemented tmp dir per user